lab_3_generate_by_ngrams/main.py

```diff
@@ -208,6 +311,18 @@ def build(self) -> int:
         In case of corrupt input arguments or methods used return None,
         1 is returned
         """
+        if not (isinstance(self._encoded_corpus, tuple) and self._encoded_corpus):
+            return 1
+
+        corpus = self._extract_n_grams(self._encoded_corpus)
+
+        if not isinstance(corpus, tuple) or len(corpus) == 0:
+            return 1
+
+        for i in set(corpus):
+            self._n_gram_frequencies.update({i: corpus.count(i)/str(corpus).count(str(i)[:-3])})
```

Comment:
what does `-3` mean?

Reply:
I wanted to remove last token and ')' but I didn't think about the fact that tokens could be longer, so I fixed it.